### ISSUE TITLE
error with network time protocols

### ISSUE THREAD

**Issue by bitchecker** [lxc/lxc#2108]
# Required information

 * Distribution: `Fedora`
 * Distribution version: `27`
 * The output of
   * lxc-start --version: `2.0.9`
   * lxc-checkconfig:
```
Kernel configuration not found at /proc/config.gz; searching...
Kernel configuration found at /boot/config-4.14.14-300.fc27.x86_64
--- Namespaces ---
Namespaces: enabled
Utsname namespace: enabled
Ipc namespace: enabled
Pid namespace: enabled
User namespace: enabled
Network namespace: enabled

--- Control groups ---
Cgroups: enabled

Cgroup v1 mount points: 
/sys/fs/cgroup/systemd
/sys/fs/cgroup/pids
/sys/fs/cgroup/net_cls,net_prio
/sys/fs/cgroup/cpuset
/sys/fs/cgroup/memory
/sys/fs/cgroup/cpu,cpuacct
/sys/fs/cgroup/hugetlb
/sys/fs/cgroup/blkio
/sys/fs/cgroup/perf_event
/sys/fs/cgroup/freezer
/sys/fs/cgroup/devices

Cgroup v2 mount points: 
/sys/fs/cgroup/unified

Cgroup v1 clone_children flag: enabled
Cgroup device: enabled
Cgroup sched: enabled
Cgroup cpu account: enabled
Cgroup memory controller: enabled
Cgroup cpuset: enabled

--- Misc ---
Veth pair device: enabled, loaded
Macvlan: enabled, not loaded
Vlan: enabled, not loaded
Bridges: enabled, loaded
Advanced netfilter: enabled, not loaded
CONFIG_NF_NAT_IPV4: enabled, loaded
CONFIG_NF_NAT_IPV6: enabled, loaded
CONFIG_IP_NF_TARGET_MASQUERADE: enabled, loaded
CONFIG_IP6_NF_TARGET_MASQUERADE: enabled, not loaded
CONFIG_NETFILTER_XT_TARGET_CHECKSUM: enabled, loadedCONFIG_NETFILTER_XT_MATCH_COMMENT: enabled, not loaded
FUSE (for use with lxcfs): enabled, loaded

--- Checkpoint/Restore ---
checkpoint restore: enabled
CONFIG_FHANDLE: enabled
CONFIG_EVENTFD: enabled
CONFIG_EPOLL: enabled
CONFIG_UNIX_DIAG: enabled
CONFIG_INET_DIAG: enabled
CONFIG_PACKET_DIAG: enabled
CONFIG_NETLINK_DIAG: enabled
File capabilities: enabled

Note : Before booting a new kernel, you can check its configuration
usage : CONFIG=/path/to/config /usr/bin/lxc-checkconfig
```
   * uname -a: `Linux wolf 4.14.14-300.fc27.x86_64 #1 SMP Fri Jan 19 13:19:54 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux`
   * cat /proc/self/cgroup:
```
11:devices:/user.slice
10:freezer:/
9:perf_event:/
8:blkio:/user.slice
7:hugetlb:/
6:cpu,cpuacct:/user.slice
5:memory:/user.slice
4:cpuset:/
3:net_cls,net_prio:/
2:pids:/user.slice/user-1000.slice/session-2.scope
1:name=systemd:/user.slice/user-1000.slice/session-2.scope
0::/user.slice/user-1000.slice/session-2.scope
```
   * cat /proc/1/mounts:
```
sysfs on /sys type sysfs (rw,nosuid,nodev,noexec,relatime)
proc on /proc type proc (rw,nosuid,nodev,noexec,relatime)
devtmpfs on /dev type devtmpfs (rw,nosuid,size=8140840k,nr_inodes=2035210,mode=755)
securityfs on /sys/kernel/security type securityfs (rw,nosuid,nodev,noexec,relatime)
tmpfs on /dev/shm type tmpfs (rw,nosuid,nodev)
devpts on /dev/pts type devpts (rw,nosuid,noexec,relatime,gid=5,mode=620,ptmxmode=000)
tmpfs on /run type tmpfs (rw,nosuid,nodev,mode=755)
tmpfs on /sys/fs/cgroup type tmpfs (ro,nosuid,nodev,noexec,mode=755)
cgroup on /sys/fs/cgroup/unified type cgroup2 (rw,nosuid,nodev,noexec,relatime)
cgroup on /sys/fs/cgroup/systemd type cgroup (rw,nosuid,nodev,noexec,relatime,xattr,name=systemd)
pstore on /sys/fs/pstore type pstore (rw,nosuid,nodev,noexec,relatime)
cgroup on /sys/fs/cgroup/pids type cgroup (rw,nosuid,nodev,noexec,relatime,pids)
cgroup on /sys/fs/cgroup/net_cls,net_prio type cgroup (rw,nosuid,nodev,noexec,relatime,net_cls,net_prio)
cgroup on /sys/fs/cgroup/cpuset type cgroup (rw,nosuid,nodev,noexec,relatime,cpuset)
cgroup on /sys/fs/cgroup/memory type cgroup (rw,nosuid,nodev,noexec,relatime,memory)
cgroup on /sys/fs/cgroup/cpu,cpuacct type cgroup (rw,nosuid,nodev,noexec,relatime,cpu,cpuacct)
cgroup on /sys/fs/cgroup/hugetlb type cgroup (rw,nosuid,nodev,noexec,relatime,hugetlb)
cgroup on /sys/fs/cgroup/blkio type cgroup (rw,nosuid,nodev,noexec,relatime,blkio)
cgroup on /sys/fs/cgroup/perf_event type cgroup (rw,nosuid,nodev,noexec,relatime,perf_event)
cgroup on /sys/fs/cgroup/freezer type cgroup (rw,nosuid,nodev,noexec,relatime,freezer)
cgroup on /sys/fs/cgroup/devices type cgroup (rw,nosuid,nodev,noexec,relatime,devices)
configfs on /sys/kernel/config type configfs (rw,relatime)
/dev/sda1 on / type ext4 (rw,relatime,data=ordered)
systemd-1 on /proc/sys/fs/binfmt_misc type autofs (rw,relatime,fd=39,pgrp=1,timeout=0,minproto=5,maxproto=5,direct,pipe_ino=16517)
mqueue on /dev/mqueue type mqueue (rw,relatime)
debugfs on /sys/kernel/debug type debugfs (rw,relatime)
hugetlbfs on /dev/hugepages type hugetlbfs (rw,relatime,pagesize=2M)
tmpfs on /tmp type tmpfs (rw,nosuid,nodev)
/dev/mapper/luks-d3d14bd7-ede6-4ad5-ad3f-729979d054c0 on /media/Multimedia type ext4 (rw,relatime,data=ordered)
sunrpc on /var/lib/nfs/rpc_pipefs type rpc_pipefs (rw,relatime)
fusectl on /sys/fs/fuse/connections type fusectl (rw,relatime)
tmpfs on /run/user/1000 type tmpfs (rw,nosuid,nodev,relatime,size=1631128k,mode=700,uid=1000,gid=1000)
gvfsd-fuse on /run/user/1000/gvfs type fuse.gvfsd-fuse (rw,nosuid,nodev,relatime,user_id=1000,group_id=1000)
```

# Issue description

I'm trying to start into a container network time syncronization but, when i try to start daemon, it fails.

# Steps to reproduce

 1. Step one
`lxc-create -n container1 -t centos -- -R 7 -a x86_64`
`lxc-start -n container1`
 2. Step two
`lxc-console -n container1`

```
# yum install -y chrony 

Loading mirror speeds from cached hostfile
 * base: it.centos.contactlab.it
 * epel: pkg.adfinis-sygroup.ch
 * extras: it.centos.contactlab.it
 * updates: it.centos.contactlab.it
Resolving Dependencies
There are unfinished transactions remaining. You might consider running yum-complete-transaction, or "yum-complete-transaction --cleanup-only" and "yum history redo last", first to finish them. If those don't work you'll have to try removing/installing packages by hand (maybe package-cleanup can help).
--> Running transaction check
---> Package chrony.x86_64 0:3.1-2.el7.centos will be installed
--> Finished Dependency Resolution

Dependencies Resolved

================================================================================================================================================================================================================================================================================
 Package                                                         Arch                                                            Version                                                                    Repository                                                     Size
================================================================================================================================================================================================================================================================================
Installing:
 chrony                                                          x86_64                                                          3.1-2.el7.centos                                                           base                                                          237 k

Transaction Summary
================================================================================================================================================================================================================================================================================
Install  1 Package

Total download size: 237 k
Installed size: 464 k
Downloading packages:
chrony-3.1-2.el7.centos.x86_64.rpm                                                                                                                                                                                                                       | 237 kB  00:00:01     
Running transaction check
Running transaction test
Transaction test succeeded
Running transaction
  Installing : chrony-3.1-2.el7.centos.x86_64                                                                                                                                                                                                                               1/1 
  Verifying  : chrony-3.1-2.el7.centos.x86_64                                                                                                                                                                                                                               1/1 

Installed:
  chrony.x86_64 0:3.1-2.el7.centos                                                                                                                                                                                                                                              

Complete!
```

 3. Step three

```
# systemctl start chronyd
 # systemctl status chronyd.service 
● chronyd.service - NTP client/server
   Loaded: loaded (/usr/lib/systemd/system/chronyd.service; enabled; vendor preset: enabled)
   Active: inactive (dead)
Condition: start condition failed at Fri 2018-01-26 10:32:45 UTC; 9s ago
           ConditionCapability=CAP_SYS_TIME was not met
     Docs: man:chronyd(8)
           man:chrony.conf(5)
```
another test can be done with `ntp`
```
# systemctl start ntpd
# systemctl status ntpd
● ntpd.service - Network Time Service
   Loaded: loaded (/usr/lib/systemd/system/ntpd.service; disabled; vendor preset: disabled)
   Active: failed (Result: exit-code) since Fri 2018-01-26 10:35:48 UTC; 1s ago
  Process: 586 ExecStart=/usr/sbin/ntpd -u ntp:ntp $OPTIONS (code=exited, status=0/SUCCESS)
 Main PID: 587 (code=exited, status=255)

Jan 26 10:35:48 ntp ntpd[587]: Listen normally on 4 eth0 fe80::216:3eff:fec3:9dda UDP 123
Jan 26 10:35:48 ntp ntpd[587]: Listen normally on 5 lo ::1 UDP 123
Jan 26 10:35:48 ntp ntpd[587]: Listening on routing socket on fd #22 for interface updates
Jan 26 10:35:48 ntp ntpd[587]: 0.0.0.0 c016 06 restart
Jan 26 10:35:48 ntp ntpd[587]: ntp_adjtime() failed: Operation not permitted
Jan 26 10:35:48 ntp ntpd[587]: 0.0.0.0 c012 02 freq_set kernel 0.000 PPM
Jan 26 10:35:48 ntp ntpd[587]: 0.0.0.0 c011 01 freq_not_set
Jan 26 10:35:48 ntp systemd[1]: ntpd.service: main process exited, code=exited, status=255/n/a
Jan 26 10:35:48 ntp systemd[1]: Unit ntpd.service entered failed state.
Jan 26 10:35:48 ntp systemd[1]: ntpd.service failed.
```

**Comment by brauner** [comment]
If you have a privileged container you need to keep  CAP_SYS_TIME for the container which is dropped by default. So put

    lxc.cap.drop =
    lxc.cap.drop = mac_admin mac_override sys_module sys_rawio

and try again

**Comment by bitchecker** [comment]
hi @brauner,
first of all thanks for your reply.

is it possible to configure these parameters for each container and not individually?

**Comment by brauner** [comment]
Yes, every option you place in files you include in `/usr/share/lxc/config/common.conf.d/` will be automatically included. So if you do something like:

     touch 01-sys-time.conf

and add

    lxc.cap.drop =
    lxc.cap.drop = mac_admin mac_override sys_module sys_rawio

to this file you should be all set.

**Comment by bitchecker** [comment]
so even the already created containers will be able to use these features in this way?

**Comment by bitchecker** [comment]
problem solved.
thank you so much.

**Comment by itreers** [comment]
This solved my problem on Proxmox 5.3 too. Thanks.

**Comment by gstanden** [comment]
This solved my problem on Orabuntu-LXC also.  Oracle RAC requires NTP to be fully configured and running on each Oracle RAC node.  Until this solution I did not have a solid way to configure NTP in each LXC Oracle RAC container, but now that problem is solved.  Thank you so much!
